Add a `noMatchSnapshots` option.

tmeasday opened the pull request:
See (https://github.com/storybooks/storyshots/pull/93). 

Issue:

The idea is that early on in a project, when things are changing a lot, snapshot tests are not particularly useful, but it's still helpful to "smoke test" all stories to ensure they run without error.

## What I did

I added a `noMatchSnapshots` option to just setup a test that runs each story, without checking snapshots.

## How to test

1. Create a project, install storyshots but pass `initStoryshots({ noMatchSnapshots: true })`.
2. Add a story, run `jest`. Should pass.
3. Change the component's output, run `jest` again. Should still pass
4. Make the component error in the render function, run `jest` again. Should fail.

## Notes:

Perhaps it would make sense to just allow passing a custom `test` function, to cover off cases like this and https://github.com/storybooks/storyshots/pull/76? Then `storyshots` is more of a generic tool to integrate storybook with Jest.

```js
initStoryshots({
  test(renderedStory) {
     // In my case, just render without looking at the result
     renderer.create(renderedStory).toJSON()

     // For #76, shallow render + snapshot
     const tree = shallowToJson(shallow(renderedStory))
     expect(tree).toMatchSnapshot()
  }
});
```

tmeasday committed:
Add a `noMatchSnapshots` option.

shilman commented on packages/storyshots/README.md:
```diff
@@ -97,3 +97,7 @@ initStoryshots({
 
 Here is an example of [a regex](https://regex101.com/r/vkBaAt/2) which does not pass if `"Relay"` is in the name: `/^((?!(r|R)elay).)*$/`.
+
+### `noMatchSnapshots`
```
@tmeasday What do you think about creating an option `compareSnapshots` that defaults to `true`? I don't like the double negative of `noMatchSnapshots`. Or just call the option you added `skipComparison`?

ndelangen replied:
I'd name this `autoUpdate`. The default is false.

shilman commented:
@tmeasday FWIW, I prefer your genericized solution to the current proposal.

shilman commented:
I'm having problems testing out the change. It's probably unrelated to your change. What I did:

```
cd storybook
npm install
cd packages/storyshots
npm link .
cd ../../examples/cra-storybook
npm link \@kadira/storyshots
emacs src/storyshots.tests.js
npm run test
```

With the following contents for `storyshots.tests.js`:

```js
import initStoryshots from 'storyshots'
initStoryshots({ noMatchSnapshts: true })
```

I get the error:
```
  ● Test suite failed to run

    Cannot find module 'react-test-renderer' from 'index.js'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:151:17)
      at Object.<anonymous> (../../../../../.nvm/versions/node/v6.2.1/lib/node_modules/storyshots/dist/index.js:13:26)
      at Object.<anonymous> (src/storyshots.test.js:1:173)
```

I tried `npm i -D react-test-renderer` to add the peer dependency, but still get the same error.

I suspect this is related to Lerna. @tmeasday @ndelangen  @aaronmcadam. Any help would be much appreciated.

ndelangen reviewed (commented):
I'd name this `autoUpdate` and have it default value be `false`.

But I'm not really feeling much for this feature myself, since updating snapshots is real easy already anyway.

tmeasday commented:
> I'm having problems testing out the change

Apologies, I didn't test the changes on the monorepo. My bad. Let me have a go at it and I'll ping you when I figure it out.

tmeasday commented:
> I'd name this autoUpdate and have it default value be false.

> But I'm not really feeling much for this feature myself, since updating snapshots is real easy already anyway.

@ndelangen -- the idea isn't to update the snapshots, it's to "smoke test" your stories but not to snapshot them.

When you are early in development, pretty much every commit changes your snapshots, and so (in my experience) you tend to disable snapshot tests as they are too noisy. However it's pretty useful to know if a commit has broken a component (in the sense that it fails to render for some story). That's the idea of this change.

@shilman -- I tend to agree that we should do the more generic thing.

tmeasday committed:
Make storybook et al peer deps.

Storyshots wants to plugin to the version used by the app, not have its own version. So its a clear peer dep.

Not sure it should depend on *storybook* at all given it is optional whether the user uses it or `react-native-storybook`

tmeasday committed:
Use storyshots in the CRA test app

tmeasday commented:
I rebased this onto https://github.com/storybooks/storybook/pull/971

Should be easy to test now @shilman.

Haven't changed the API yet though, LMK if it works for you and I'll look at that.

ndelangen committed:
Merge branch 'master' into storyshots-no-match-snapshots

tmeasday commented:
Closing this in favour of https://github.com/storybooks/storybook/issues/1034, simple PR to follow
